### PR TITLE
Add top_k_accuracy_score

### PR DESCRIPTION
**Pull request by msluszniak**

### CODE REVIEW

File: lib/scholar/metrics.ex [hunk 1]
```diff
@@ -1205,6 +1226,100 @@ defmodule Scholar.Metrics do
     1 - Nx.sum(weights_matrix * cm) / Nx.sum(weights_matrix * expected)
   end
 
+  @doc """
+
```
Comment (josevalim) [inline lib/scholar/metrics.ex:1230]:
```suggestion
```

File: lib/scholar/metrics.ex [hunk 2]
```diff
@@ -1205,6 +1226,100 @@ defmodule Scholar.Metrics do
...
+  Top-k Accuracy classification score.
+
+  This metric computes the number of times where the correct label is among the top k labels predicted (ranked by predicted scores).
```
Comment (josevalim) [inline lib/scholar/metrics.ex:1233]:
```suggestion
  This metric computes the number of times where the correct label is
  among the top k labels predicted (ranked by predicted scores).
```

File: lib/scholar/metrics.ex [hunk 3]
```diff
@@ -1205,6 +1226,100 @@ defmodule Scholar.Metrics do
...
+
+      iex> y_true = Nx.tensor([0, 1, 0, 1, 0])
+      iex> y_score = Nx.tensor([0.55, 0.3, 0.1, -0.2, 0.99])
+      iex> Scholar.Metrics.top_k_accuracy_score(y_true, y_score, k: 1, num_classes: 2)
+      #Nx.Tensor<
+        f32
+        0.20000000298023224
+      >
+  """
+
```
Comment (josevalim) [inline lib/scholar/metrics.ex:1267]:
```suggestion
```

File: lib/scholar/metrics.ex [hunk 4]
```diff
@@ -1205,6 +1226,100 @@ defmodule Scholar.Metrics do
...
+          Nx.any(Nx.new_axis(y_true, 0) == Nx.transpose(sorted_pred[[.., 0..(k - 1)//1]]),
+            axes: [0]
+          )
+      end
+
+    case opts[:normalize] do
+      true -> Nx.mean(hits)
+      false -> Nx.sum(hits)
+    end
```
Comment (josevalim) [inline lib/scholar/metrics.ex:1307]: Can this function be used anywhere? https://hexdocs.pm/nx/Nx.html#top_k/2

Comment (msluszniak) [reply lib/scholar/metrics.ex:1306]: If this function allows us to choose the axis then yes. But currently, I need to use `Nx.argsort` with the axis set to 1.

https://github.com/msluszniak/scholar/blob/9854fbc25a9fb9cddf0947aa4113c51639cf74f9/lib/scholar/metrics.ex#L1295C1-L1300

**Review by josevalim — APPROVED** [review]

**Pull request merged**